hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        ori_cov = OriginalCoverage.objects.get(id=element_id)
+        if ori_cov:
+            # # update projection string info
+            for key, value in kwargs.iteritems():
+                if key in ('projection_string','projection_name','datum','unit', 'northlimit', 'southlimit', 'eastlimit', 'westlimit'):
+                    setattr(ori_cov, key, value)
+                    ori_cov.save()
+        else:
+            raise ObjectDoesNotExist("No coverage element was found for the provided id: %d" % element_id)
+
+    @classmethod
+    def remove(cls, element_id):
+        ori_cov = OriginalCoverage.objects.get(id=element_id)
+        if ori_cov:
+            ori_cov.delete()
+        else:
+            raise ObjectDoesNotExist("No original coverage element exists for id: %d."%element_id)
+
+
+class FieldInformation(AbstractMetaDataElement):
+    term = 'FieldInformation'
+
+    fieldName = models.CharField(max_length=128, null=False, blank=False)
+    fieldType = models.CharField(max_length=128, null=False, blank=False)
+    fieldTypeCode = models.CharField(max_length=50, null=True, blank=True)
+    fieldWidth = models.IntegerField(null=True, blank=True)
+    fieldPrecision = models.IntegerField(null=True, blank=True)
+
+    def __unicode__(self):
+        self.fieldName
+    @classmethod
+    def create(cls, **kwargs):
+        if 'fieldName' in kwargs:
+            if not 'fieldType' in kwargs:
+                raise ValidationError("fieldType of FieldInformation is missing.")
+
+            metadata_obj = kwargs['content_object']
+            field_info = FieldInformation.objects.create(fieldName = kwargs['fieldName'], fieldType=kwargs['fieldType'],content_object=metadata_obj,)
+
+            for key, value in kwargs.iteritems():
+                if key in ('fieldTypeCode', 'fieldWidth', 'fieldPrecision'):
+                    setattr(field_info, key, value)
+                    field_info.save()
+            return field_info
+        else:
+            raise ValidationError("fieldName of FieldInformation is missing.")
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        field_info = FieldInformation.objects.get(id=element_id)
+        if field_info:
+            for key, value in kwargs.iteritems():
+                if key in ('fieldName', 'fieldTypeCode', 'fieldType', 'fieldWidth', 'fieldPrecision'):
+                    setattr(field_info, key, value)
+            field_info.save()
+        else:
+            raise ObjectDoesNotExist("No FieldInformation element can be found for the provided id: %s" % kwargs['id'])
+
+    @classmethod
+    def remove(cls, element_id):
+        field_info = FieldInformation.objects.get(id=element_id)
+        if field_info:
+            field_info.delete()
+        else:
+            raise ValidationError("FieldInformation element of the geographic feature resource cannot be deleted: %d."%element_id)
```

Comment:
Sorry but I'm going to insist on these redundancies every time they appear in this change set and fact that I can repeat the same feedback so many times is just more evidence of the redundant nature of these classes.